source/guides/first-steps/task-script.md

```diff
@@ -1,4 +1,127 @@
-Your First Task Script (NOT YET WRITTEN)
+Your First Task Script
 ----------------------
 
-**TODO: Write-up that guides the user through writing a basic task script (just a task with a narrate line) using VS-Code, emphasizing things like file extension and placement. Also run it in-game with `/ex run`.**
+<!-- **TODO: Write-up that guides the user through writing a basic task script (just a task with a narrate line) using VS-Code, emphasizing things like file extension and placement. Also run it in-game with `/ex run`.** -->
+
+```eval_rst
+.. contents:: Table of Contents
+    :local:
+```
+
+## Task script basics
+
+A task script is a stand-alone script that can be run either via in-game command with the `/ex` command, as explained in [the guide to the /ex command](https://guide.denizenscript.com/guides/first-steps/ex-command.html).
+
+Task scripts will run all of the Denizen commands that they include. A task script can be simple, it can be complicated, and it can even run other task scripts. This is useful to create logic-based script chains, like dialogue options or random content.
+
+## Task script syntax
+
+Here's an example of a basic task script.
+
+```dscript_green
+example_task:
+    type: task
+    script:
+    - narrate "This is a basic task script!"
+```
+
+This script will narrate the text:
+
+```
+This is a basic task script!
+```
+
+to the player attached to the script. If you use the `/ex` command to run this script - specifically, by running `/ex run example_task` in-game, *you* will be the attached player.
+
+You can also run a task script and manually specify the attached player. For example, to run this script for the player named `mcmonkey`, you would run this command: `/ex run example_task player:<server.match_player[mcmonkey]>`
+
+The player is found by their username with the `<server.match_player[]>` tag, which returns a PlayerTag of the player whose username matches the text you put between the `[]` brackets. If this is confusing, don't worry! We'll talk more about tags later on - this is just an explanatory preview to give you an idea of what's coming. You can also return to it later on once you understand more about these concepts.
```

Comment:
All of this stuff about tags should not be on this page.

Reply:
I think "feature previews" like this are helpful, especially for users already familiar with scripting who may otherwise feel like the guides are too dumbed-down for them at the early stages.

Might I suggest that we flag such sections instead, so that newer users know they can be skipped?

Reply:
Mentioning an upcoming feature? Sure! Describing in such detail the feature and how it works? No.